docs/debugging.rst

```diff
@@ -123,6 +123,17 @@ It is similar to Python-scope ``print``.
 Unlike ``print``, ``ti.static_print`` will only print the expression once at compile-time, and
 therefore it has no runtime cost.
 
+Serial execution
+----------------
+
+The automatic parallelization feature of Taichi may lead to undeterministic behaviors.
+For debugging purposes, it may be useful to serialize program execution to get repeatable results and to diagnose data races.
+When running your Taichi program on CPUs,
+you can initialize Taichi to use a single thread using ``cpu_max_num_threads=1``, so that the whole program becomes serial and deterministic. For example,
+
+``ti.init(arch=ti.cpu, cpu_max_num_threads=1)``
```

Comment:
May also worth mention ti.cc is always serialized for people who have it? It actually have better performance than serialized llvm cpu on my end.

Reply:
That's a good point. However, given `ti.cc` is so far Linux-only, we may want to stick to the more portable LLVM CPU backend to keep things simple here. Also note that the LLVM cpu backend supports all the extensions. Switching to `ti.cc` may confuse people since their code may no longer run due to the lack of certain extensions...